### PR TITLE
fix(nuxi): restart nuxt when `distDir` is unlinked

### LINKED ISSUE #15069: cannot find client.manifest.mjs when adding or removing package while dev server is up
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.17.0`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `yarn@3.2.3`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

- start nuxt dev 
- on another terminal , install any package using yarn (not tested with npm or pnpm)
- try to go on any page 
- error 500 

### Describe the bug

Hi :wave: ,

adding a package while the dev server is on cause `.nuxt/dev` and `.nuxt/dist` being deleted.

stopping and reloading the server solves the issue since it regenerate `.nuxt` directory.

I reproduced this on window and WSL(ubuntu)

### Additional context

_No response_

### Logs

```shell
Cannot find module '/mnt/d/GIT/test/.nuxt/dist/server/client.manifest.mjs' imported from /mnt/d/GIT/test/.nuxt/dev/index.mjs
```

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve https://github.com/nuxt/nuxt/issues/15069

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Hi :wave: 
this adds a watcher on `.nuxt/dist`. The dev server would reload nuxt by itself if the `dist` directory has been removed. 
This would avoid to manually restart the server.

Use case: 
- Start dev server 
- "Oh i forgot to add another package"
- `npm install another-package` in another terminal 
- `postinstall` is run with `nuxi prepare`, clear buildDir
- dev server break :cry:                                         -- dev server restart 👍 
- `ctrl` + C, ⬆️ , `enter`, wait for server to restart --- nuxt ready

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/commands/dev.ts [hunk 1]
```diff
@@ -158,6 +158,11 @@ export default defineNuxtCommand({
 
     await load(false)
 
+    const distWatcher = chokidar.watch(resolve(rootDir, '.nuxt/dist'), { ignoreInitial: true })
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:161]: I think we can use nuxt.options.buildDir + dist

**Review by danielroe — COMMENTED** [review]
What a great idea!

File: packages/nuxi/src/commands/dev.ts (hunk 1 above)
Comment (huang-julien) [reply packages/nuxi/src/commands/dev.ts:161]: yes :+1:

File: packages/nuxi/src/commands/dev.ts [hunk 2]
```diff
@@ -75,11 +77,20 @@ export default defineNuxtCommand({
...
+        if (distWatcher) {
+          distWatcher.close()
+        }
+
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
         if (!isRestart) {
           showURL()
         }
 
+        distWatcher = chokidar.watch(resolve(currentNuxt.options.buildDir, 'dist'), { ignoreInitial: true })
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:89]: shall we set `depth: 0` to only watch the directory itself?

**Review by danielroe — APPROVED** [review]
This is a very nice fix.

**Review by pi0 — APPROVED** [review]
Good idea! We might extend/change it later to other dirs such as `.cache` since we not always remove top level dist

**Pull request merged**